### DISCUSSION TITLE
Support zero queue consumer for partitioned topic.

### DISCUSSION THREAD

**Discussion by codelipenghui** [apache/pulsar#18846] [General]
**Is your feature request related to a problem? Please describe.**
Currently, the zero queue consumer just can subscribe to a non-partitioned topic. In some case, we need to use zero queue consumer to subscribe to a partitioned topic.

**Comment by merlimat** [comment]
It’s not enabled because we cannot know which partition the next message will be coming from. Any suggestion on how to achieve that?

**Comment by sundar-10** [comment]
Hello, I am planning to work on this issue. I am planning to read how zero queue consumer is subscribed to a nonpartitioned topic and proceed from there by getting some kind of input for which partition this consumer should be subscribed to and so on. Can you tell me where to find the code related to this?

**Comment by codelipenghui** [comment]
@sundar-10 You can check the `ZeroQueueConsumerImpl`.

**Comment by codelipenghui** [comment]
@merlimat How about syncing the backlog between the broker and client periodically. Treat topics with more backlogs as a high priority. Since we don't know if a topic will there be any messages written, to avoid a consumer block to one topic(no more data are written), we can reset the permits in a sync cycle if there are topics that have more backlog, looks like 

1. reset permits to 0 
2. re-order the consumers 
3. trigger read data from the high priority consumer

Users can achieve the desired purpose by setting the frequency of synchronization.

We can optimize the sync mechanism, e.g. carry the backlog when dispatching messages. Only if the message is not dispatched for a period of time, perform a separate sync command.

**Comment by leizhiyuan** [comment]
what is the current status?
